base/applications/rapps/unattended.cpp

```diff
@@ -104,6 +121,98 @@ HandleSetupCommand(CAppDB *db, LPWSTR szCommand, int argcLeft, LPWSTR *argvLeft)
     return DownloadListOfApplications(Applications, TRUE);
 }
 
+static BOOL
+HandleUninstallCommand(CAppDB &db, UINT argcLeft, LPWSTR *argvLeft)
+{
+    UINT argi = 0, silent = FALSE, byregkeyname = FALSE;
+    for (; argcLeft; ++argi, --argcLeft)
+    {
+        if (!StrCmpIW(argvLeft[argi], L"/S"))
+            ++silent;
+        else if (!StrCmpIW(argvLeft[argi], L"/K"))
+            ++byregkeyname;
+        else
+            break;
+    }
+    if (argcLeft != 1)
+        return FALSE;
+
+    CStringW buf;
+    LPCWSTR name = argvLeft[argi];
+    BOOL retval = FALSE;
+    CAppInfo *pInfo = NULL, *pDelete = NULL;
+
+    if (!byregkeyname)
+    {
+        for (UINT i = ENUM_INSTALLED_MIN; !pInfo && i <= ENUM_INSTALLED_MAX; ++i)
+        {
+            pInfo = SearchForAppWithDisplayName(db, AppsCategories(i), name);
+        }
+
+        if (!pInfo)
+        {
+            CAvailableApplicationInfo *p = db.FindAvailableByPackageName(name);
+            if (p)
+            {
+                CConfigParser *cp = p->GetConfigParser();
+                if (cp && cp->GetString(DB_REGNAME, buf) && !buf.IsEmpty())
+                {
+                    name = buf.GetString();
+                    byregkeyname = TRUE;
+                }
+            }
+        }
+    }
+
+    if (byregkeyname)
+    {
+        // Force a specific key type if requested (<M|U>[32|64]<\\KeyName>)
+        if (name[0])
+        {
+            REGSAM wow = 0;
+            UINT i = 1;
+            if (name[i] == '3' && name[i + 1])
+                wow = KEY_WOW64_32KEY, i += 2;
+            else if (name[i] == '6' && name[i + 1])
+                wow = KEY_WOW64_64KEY, i += 2;
+
+            if (name[i++] == '\\')
+            {
+                pInfo = CAppDB::CreateInstalledAppInstance(name + i, name[0] == 'U', wow);
+            }
+        }
+
+        if (!pInfo)
+        {
+            pInfo = CAppDB::CreateInstalledAppByRegistryKey(name);
+        }
+        pDelete = pInfo;
+    }
+
+    if (pInfo)
+    {
+        retval = pInfo->UninstallApplication(silent ? UCF_SILENT : 0);
+    }
+    delete pDelete;
+    return retval;
+}
+
+static BOOL
+HandleGenerateInstallerCommand(CAppDB &db, UINT argcLeft, LPWSTR *argvLeft)
+{
+    extern BOOL
+    ExtractAndRunGeneratedInstaller(const CAvailableApplicationInfo &AppInfo, LPCWSTR Archive);
```

Comment:
Please don't forward declare functions inside other functions, unless there is a good reason to do so.

Reply:
This is the only place it will be called, I don't know if you count that as a good reason.